File: NetCoreCv.Core/Models/CurriculumVitae.cs
using Microsoft.AspNetCore.Identity;

namespace NetCoreCv.Core.Models;

public class CurriculumVitae
{
    public int CurriculumVitaeId { get; set; }

    public string? UserId { get; set; }

    public IdentityUser? User { get; set; }

    public int? CurrentAddressId { get; set; }

    public Address? CurrentAddress { get; set; }

    public IEnumerable<Education>? Education { get; set; }

    public IEnumerable<WorkExperience>? WorkExperience { get; set; }

    public IEnumerable<Project>? Projects { get; set; }
}

File: client/src/schemas/curriculum-vitae.ts
import { Address } from "./address";
import { Education } from "./education";
import { WorkExperience } from "./work-experience";
import { Project } from "./project";

export class CurriculumVitae {
    id: number;
    curriculumVitaeId: number;
    userName: string;
    emailAddress: string;
    currentAddressId: number | null;
    currentAddress: Address | null;
    education: Education[];
    workExperience: WorkExperience[];
    projects: Project[];
}